metrics.go

```diff
@@ -0,0 +1,52 @@
+package ipam
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	prometheusNamespace = "ipam"
```

Comment:
As a note, I didn't see much point setting a prometheus subsystem here, the only thing that would make sense is `ipam`, which just means we have a crappy metric name.